### PR TITLE
Add Dealfront source connector

### PR DESCRIPTION
**Pull request by mkrawc**
Add Dealfront connector

### CODE REVIEW

**Comment by mkrawc** [comment]
[Airbyte > Dealfront API Custom Connector](https://trello.com/c/PV2kk9GQ/8607-airbyte-dealfront-api-custom-connector)

File: airbyte-integrations/connectors/source-dealfront/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+[build-system]
+requires = [ "poetry-core>=1.0.0",]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+version = "0.1.0"
+name = "source-dealfront"
+description = "Source implementation for dealfront."
+authors = [ "Airbyte <contact@airbyte.io>",]
```
Comment (dzidas) [inline airbyte-integrations/connectors/source-dealfront/pyproject.toml:9]: i know it's a minor thing but maybe we could use our CorralData emeils :)

**Review by dzidas — APPROVED** [review]

File: airbyte-integrations/connectors/source-dealfront/source_dealfront/components.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import requests
+from airbyte_cdk.sources.declarative.decoders.decoder import Decoder
+from airbyte_cdk.sources.declarative.decoders.json_decoder import JsonDecoder
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.types import Config
+
+
+@dataclass
+class UnnestingDpathExtractor(RecordExtractor):
```
Comment (mkrawc) [inline airbyte-integrations/connectors/source-dealfront/source_dealfront/components.py:18]: This was implemented based on https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/dpath_extractor.py

File: airbyte-integrations/connectors/source-dealfront/source_dealfront/components.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        else:
+            extracted = []
+
+        if self.unnest_extractor:
+            return self.unnest_fields(extracted)
+
+        return extracted
+
+    def unnest_fields(self, records: List[Mapping[str, Any]]) -> List[Mapping[str, Any]]:
+        if isinstance(records, list):
```
Comment (dzidas) [inline airbyte-integrations/connectors/source-dealfront/source_dealfront/components.py:96]: what about this check first?
```if not isinstance(records, list):
            return self.unnest_extractor.extract_from_json(records)```

File: airbyte-integrations/connectors/source-dealfront/source_dealfront/components.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            unnest = []
+            for record in records:
+                for unnest_record in self.unnest_extractor.extract_from_json(record):
+                    merged = dict()
+                    merged.update(record)
+                    merged.update(unnest_record)
+                    unnest.append(merged)
+            return unnest
```
Comment (dzidas) [inline airbyte-integrations/connectors/source-dealfront/source_dealfront/components.py:104]: for this GPT suggested me this list comprehension :)

```
# Use list comprehension for more compact and Pythonic code
        unnested_records = [
            {**record, **unnest_record}
            for record in records
            for unnest_record in self.unnest_extractor.extract_from_json(record)
        ]
        return unnested_records
```

**Pull request merged**